agent/src/agent_manager.rs

```diff
@@ -139,38 +141,37 @@ impl AgentManager {
         }
     }
 
-    async fn store_and_forward_own_workload_states(&mut self, to_server_msg: ToServer) {
-        log::debug!("Storing and forwarding own workload states.");
-
-        let ToServer::UpdateWorkloadState(common::commands::UpdateWorkloadState {
-            workload_states,
-        }) = to_server_msg
-        else {
-            std::unreachable!("expected UpdateWorkloadState msg.");
-        };
+    async fn store_and_forward_own_workload_states(
+        &mut self,
+        mut new_workload_state: WorkloadState,
+    ) {
+        // execute hysteresis on the local workload states as we could be stopping
+        // [impl->swdd~agent-manager-hysteresis_on-workload-states-of-its-workloads~1]
+        if let Some(old_execution_state) = self
+            .workload_state_store
+            .get_state_of_workload(new_workload_state.instance_name.workload_name())
+        {
+            new_workload_state.execution_state =
+                old_execution_state.transition(new_workload_state.execution_state);
+        }
 
-        workload_states.iter().for_each(|new_workload_state| {
-            log::info!(
-                "The agent '{}' reports workload state '{:?}' for the workload '{}'",
-                new_workload_state.instance_name.agent_name(),
-                new_workload_state.execution_state,
-                new_workload_state.instance_name.workload_name(),
-            );
+        log::debug!(
+            "Storing and forwarding local workload state '{:?}'.",
+            new_workload_state
+        );
```

Comment:
Shall we keep the info log (the agent reports workload state....) instead of the debug log?